src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -529,30 +554,46 @@ export class StudyViewPageStore
             )
         );
 
+        // Add special gene panel charts for each molecular profile
+        this.reactionDisposers.push(
+            reaction(
+                () => this.molecularProfileNameSet.isComplete,
+                isComplete => {
+                    if (isComplete) {
+                        const molecularProfileNameSet = this
+                            .molecularProfileNameSet.result!;
+                        for (const profileId of Object.keys(
+                            molecularProfileNameSet
+                        )) {
+                            const profileName =
+                                molecularProfileNameSet[profileId];
+                            const chartMeta = {
+                                uniqueKey: getGenePanelChartUniqueKey(
+                                    profileId
+                                ),
+                                displayName: 'Gene Panels: ' + profileName,
+                                description: '',
+                                chartType: ChartTypeEnum.PIE_CHART,
+                                dataType: ChartMetaDataTypeEnum.GENOMIC,
+                                patientAttribute: false,
+                                dimension: {
+                                    w: 1,
+                                    h: 1,
+                                },
+                                priority: 999,
+                                renderWhenDataChange: false,
+                            };
+                            this.setCustomCharts(chartMeta);
```

Comment:
i assume that if it tries to add an already existing chart type, it will just lay off?

Reply:
If the chart specified by `chartMeta` already exists in `chartTypes`, then this function will override that as well as whatever is stored in `this._customCharts` for the given key. I didn't write the function (I just extracted the code into its own function), so I don't think I should mess with it